Cloud common - Platform-as-a-Service (PaaS) closes #186 (#461)
bob-crutchley/courseware: topics/cloud/modules/platform-as-a-service/README.md

# Platform as a Service (PaaS)

<!--TOC_START-->
### Contents
	- [Description](#description)
	- [Business scenarios](#business-scenarios)
	- [Advantages](#advantages)
	- [Tasks](#tasks)

<!--TOC_END-->
### Description

> Platform as a Service is an environment in the cloud with all the resources you need for development or deployment of cloud based applications.

Reviewer: ```suggestion
**Platform as a Service (PaaS)** is an environment in the Cloud with all the resources you need for development or deployment of Cloud-based applications.
```

The resources are purchased on pay-as-you go basis, where access to them is granted through a secure internet connection.

Service is mainly targeting developers.

> **PaaS** includes *IaaS* services with the addition of: 

Reviewer: ```suggestion
**PaaS** is a stage above **Infrastructure-as-a-Service (IaaS)** - it includes all *IaaS* services with the addition of: 
```

- **middleware**
- **development tools**
- **business intelligence services**
- **database management systems**

> With *PaaS* you would avoid the hassle of paying for and managing things like:

Reviewer: ```suggestion
With *PaaS* you would avoid the hassle of paying for and managing:
```

- **software licenses**
- **application infrastructure**
- **middleware**
- **container orchestration**
- **development tools**

> The main responsibilities that are left in the users hands are the application and services that you're developing while everything else is the responsibility of the cloud provider. 

Reviewer: ```suggestion
The applications and services that are being developed remain the responsibility of the user, while everything else is the responsibility of the Cloud provider. 
```

![PaaS](https://imgur.com/7cYkMDe.jpg)

### Business scenarios

Typical use case scenarios for *PaaS* are:
- **development framework** - a framework is provided for the development of cloud applications. 
Applications can be created through the built in software components. 

> Features like: scalability, multi-tenancy, high-availability are included and managed by PaaS.

Reviewer: ```suggestion
Features such as: scalability, multi-tenancy, and high-availability are included and managed by PaaS.
```

- **business intelligence and analytics** - tools are provided for data mining, finding patterns, investment returns and for many more business related needs.

> - **additional services** - additionally there are services that provide advice in regards to: networking, security, scheduling etc. 

Reviewer: ```suggestion
- **additional services** - additionally there are services that provide advice in regards to networking, security, scheduling, etc. 
```

### Advantages

> As **PaaS** is build on top of _IaaS_ it offers all of the latter advantages with the additional ones like:

Reviewer: ```suggestion
As **PaaS** is built on top of _IaaS_ it offers all of the advantages of the latter, as well as several of its own:
```

- improving coding efficiency through the use of in-built tools
- development options for multiple platforms 

> - sophisticated tools for development, business intelligence and analytics through the pay-as-you go model allow even the smallest of companies to take advantage of the tools and save time

Reviewer: ```suggestion
- sophisticated tools for development, business intelligence and analytics through the pay-as-you go model allows for even the smallest companies to utilise them and therefore save time
```

> - remote team work is available as all the resources are reachable over the internet

Reviewer: ```suggestion
- remote teamwork is available, as all the resources are accessible over the internet
```

> - environment and all the capabilities are there for a complete application lifecycle: build, test, deploy

Reviewer: ```suggestion
- all the capabilities which exist in a typical development environment lifecycle are included - namely building, testing, and deployment```

# Platform as a Service (PaaS)

<!--TOC_START-->
## Contents
	- [Contents](#contents)
	- [Description](#description)
	- [Business scenarios](#business-scenarios)
	- [Advantages](#advantages)
	- [Tasks](#tasks)

<!--TOC_END-->
### Description

**Platform as a Service (PaaS)** is an environment in the Cloud with all the resources you need for development or deployment of Cloud-based applications.

The resources are purchased on pay-as-you go basis, where access to them is granted through a secure internet connection.

> Service is mainly targeting developers.

Reviewer: ```suggestion
*PaaS* is mainly targeted at developers.
```

**PaaS** is a stage above **Infrastructure-as-a-Service (IaaS)** - it includes all *IaaS* services with the addition of: 
- **middleware**
- **development tools**
- **business intelligence services**
- **database management systems**

With *PaaS* you would avoid the hassle of paying for and managing:
- **software licenses**
- **application infrastructure**
- **middleware**
- **container orchestration**
- **development tools**

The applications and services that are being developed remain the responsibility of the user, while everything else is the responsibility of the Cloud provider. 

![PaaS](https://imgur.com/7cYkMDe.jpg)

### Business scenarios

Typical use case scenarios for *PaaS* are:

> - **development framework** - a framework is provided for the development of cloud applications. 

Reviewer: ```suggestion
- **development framework** - a framework is provided for the development of Cloud applications. 
```
